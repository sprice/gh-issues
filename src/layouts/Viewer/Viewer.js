import React from "react"
import PropTypes from "prop-types"
import ViewerHeader from "../../components/ViewerHeader"
import IssueFilter from "../../components/IssueFilter"
import Card from "../../components/Card"
import Loading from "../../components/Loading"
import { Consumer } from "../../SearchContext"

class Viewer extends React.Component {
  componentDidMount() {
    // If there is a full page load on this page we need to load up
    // some state based on URL params.
    if (!this.props.searchParams.githubUrl) {
      this.props.searchParams.getIssues(this.props.owner, this.props.repo)
      if (
        this.props.visibilityFilter !== this.props.searchParams.visibilityFilter
      ) {
        this.props.searchParams.updateStateFromUrl(
          this.props.owner,
          this.props.repo,
          this.props.visibilityFilter
        )
      }
    }
  }

  render() {
    const error = this.props.searchParams.error
    if (this.props.searchParams.loading) {
      return <Loading githubUrl={this.props.searchParams.githubUrl} />
    }
    return (
      <div className="viewer-page">
        <ViewerHeader githubUrl={this.props.searchParams.githubUrl} />
        <IssueFilter
          getIssues={this.props.searchParams.getIssues}
          owner={this.props.owner}
          repo={this.props.repo}
          visibilityFilter={this.props.visibilityFilter}
          clearState={this.props.searchParams.clearState}
        />
        {error ? <h2 className="error">{error.message}</h2> : null}
        <div className="cards">
          {!error &&
            this.props.searchParams.issues.map(issue => {
              const isClosed = !!issue.closed_at
              const isPullRequest = !!issue.pull_request
              return (
                <Card
                  isClosed={isClosed}
                  isPullRequest={isPullRequest}
                  key={issue.number}
                >
                  <Card.Icons />
                  <Card.Title>{issue.title.substring(0, 60)}</Card.Title>
                  <Card.Body>{issue.body.substring(0, 100)}</Card.Body>
                  {issue.labels.map(label => (
                    <Card.Label key={label.id}>{label.name}</Card.Label>
                  ))}
                </Card>
              )
            })}
        </div>
      </div>
    )
  }
}

Viewer.propTypes = {
  searchParams: PropTypes.shape({
    getIssues: PropTypes.func.isRequired,
    visibilityFilter: PropTypes.string.isRequired,
    loading: PropTypes.bool.isRequired,
    githubUrl: PropTypes.string.isRequired,
    updateStateFromUrl: PropTypes.func.isRequired,
    issues: PropTypes.array.isRequired,
    clearState: PropTypes.func.isRequired,
    error: PropTypes.object
  }).isRequired,
  owner: PropTypes.string,
  repo: PropTypes.string,
  visibilityFilter: PropTypes.string
}

export default function ViewerWithContext(props) {
  return (
    <Consumer>
      {context => <Viewer {...props} searchParams={context} />}
    </Consumer>
  )
}
