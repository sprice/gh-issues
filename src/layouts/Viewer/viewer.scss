h2.error {
  text-align: center;
}
