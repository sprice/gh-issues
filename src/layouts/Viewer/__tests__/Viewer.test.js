import React from "react"
import { create } from "react-test-renderer"
import Viewer from "../Viewer"

test("Snapshop of Viewer", () => {
  const c = create(<Viewer owner="ethereum" repo="solidity" />)
  expect(c.toJSON()).toMatchSnapshot()
})
