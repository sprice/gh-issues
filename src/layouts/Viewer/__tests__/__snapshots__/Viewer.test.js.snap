// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshop of Viewer 1`] = `
<div
  className="viewer-page"
>
  <header>
    <h2>
      Github Issue Viewer
    </h2>
    <h4
      className="regular"
    >
      
    </h4>
  </header>
  <div
    className="issue-filter"
  >
    <div
      className="filters"
    >
      <a
        className=""
        href="/viewer/ethereum/solidity/all"
        onClick={[Function]}
      >
        All Issues
      </a>
      <a
        className=""
        href="/viewer/ethereum/solidity/open"
        onClick={[Function]}
      >
        Open Issues
      </a>
      <a
        className=""
        href="/viewer/ethereum/solidity/closed"
        onClick={[Function]}
      >
        Closed Issues
      </a>
      <a
        className=""
        href="/viewer/ethereum/solidity/pull_request"
        onClick={[Function]}
      >
        Pull Requests
      </a>
    </div>
    <div
      className="close-viewer"
    >
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        <i
          className="icon close"
        />
      </a>
    </div>
  </div>
  <div
    className="cards"
  />
</div>
`;
